Use list instead of Bitmapset for partitioned matview status tracking.

avamingli opened the pull request:
The original Bitmapset implementation uses int parameters, which cannot handle Oid values(unsigned int) larger than maximum positive int values. This caused errors when passing certain relation IDs.

Fix by switching to OID lists for storage. While this may be slightly less efficient for deduplication than Bitmapset, it provides correct behavior for all OID values.

The change affects how we maintain materialized view status information for partitioned tables.

Authored-by: Zhang Mingli avamingli@gmail.com

<!-- Thank you for your contribution to Apache Cloudberry (Incubating)! -->

Fixes #ISSUE_Number

### What does this PR do?
<!-- Brief overview of the changes, including any major features or fixes -->

### Type of Change
- [ ] Bug fix (non-breaking change)
- [ ] New feature (non-breaking change)
- [ ] Breaking change (fix or feature with breaking changes)
- [ ] Documentation update

### Breaking Changes
<!-- Remove if not applicable. If yes, explain impact and migration path -->

### Test Plan
<!-- How did you test these changes? -->
- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] Passed `make installcheck`
- [ ] Passed `make -C src/test installcheck-cbdb-parallel`

### Impact
<!-- Remove sections that don't apply -->
**Performance:**
<!-- Any performance implications? -->

**User-facing changes:**
<!-- Any changes visible to users? -->

**Dependencies:**
<!-- New dependencies or version changes? -->

### Checklist
- [ ] Followed [contribution guide](https://cloudberry.apache.org/contribute/code)
- [ ] Added/updated documentation
- [ ] Reviewed code for security implications
- [ ] Requested review from [cloudberry committers](https://github.com/orgs/apache/teams/cloudberry-committers)

### Additional Context
<!-- Any other information that would help reviewers? Remove if none -->

### CI Skip Instructions
<!--
To skip CI builds, add the appropriate CI skip identifier to your PR title.
The identifier must:
- Be in square brackets []
- Include the word "ci" and either "skip" or "no"
- Only use for documentation-only changes or when absolutely necessary
-->

---
<!-- Join our community:
- Mailing list: [dev@cloudberry.apache.org](https://lists.apache.org/list.html?dev@cloudberry.apache.org) (subscribe: dev-subscribe@cloudberry.apache.org)
- Discussions: https://github.com/apache/cloudberry/discussions -->

gfphoenix78 commented on src/backend/executor/nodeModifyTable.c:
```diff
@@ -1053,7 +1053,7 @@ ExecInsert(ModifyTableState *mtstate,
 	{
 		mtstate->mt_leaf_relids_inserted =
-			bms_add_member(mtstate->mt_leaf_relids_inserted, RelationGetRelid(resultRelationDesc));
+			list_append_unique_oid(mtstate->mt_leaf_relids_inserted, RelationGetRelid(resultRelationDesc));
```
If we insert N tuples into the partition table that has M children, the time may be O(N*M/nsegs). How will be the performance affected?

avamingli commented:
@my-ship-it @gfphoenix78 

I have reverted to using Bitmapset for better performance, and fixed another bug related to partitioned tables with materialized views at commit 1408e14e9cbf71e4a73af98c9fd04f59b6922303.
